File: Java/ListaExerciciosJavaPOO/Ex004/src/main/java/com/mycompany/ex004/Elevador.java
package com.mycompany.ex004;

public class Elevador {

    private int andarAtual = 0;
    private int totalAndares = 0;
    private int capacidade = 0;
    private int lotacao = 0;

    public void inicializa(int capacidade, int totalAndares) {
        this.totalAndares = totalAndares;
        this.capacidade = capacidade;
    }

    public void entra(int qtdPessoas) {
        if (qtdPessoas <= (this.capacidade - this.lotacao)) {
            System.out.println("Entrou " + qtdPessoas + " no elevador.");
            this.lotacao += qtdPessoas;
        } else if ((this.capacidade - this.lotacao) == 0) {
            System.out.println("Sem espaço, elevador cheio.");
        } else {
            System.out.println("Sem espaço, pode-se entrar " + (this.capacidade - this.lotacao) + " pessoas.");
        }
    }

    public void sai(int qtdPessoas) {
        if (qtdPessoas <= this.lotacao) {
            System.out.println("Saiu " + qtdPessoas + " pessoas.");
            this.lotacao -= qtdPessoas;
        } else if (qtdPessoas > this.lotacao) {
            System.out.println("Sem pessoas suficiente no elevador.");
        }
    }

    public void sobe(int subir) {
        if (andarAtual == totalAndares) {
            System.out.println("Já está no último andar.");
        } else if (subir > totalAndares) {
            System.out.println("Só temos " + totalAndares + " andares!");
        } else if ((andarAtual + subir) > totalAndares) {
            System.out.println("Não temos tudo isso de andares.");
        } else {
            andarAtual += subir;
            System.out.println("Subiu para o andar " + andarAtual);
        }

    }

    public void desce(int descer) {
        if (andarAtual == 0) {
            System.out.println("Já estamos no térreo.");
        } else if (descer <= andarAtual) {
            andarAtual -= descer;
            System.out.println("Desceu para o andar " + andarAtual);
        } else {
            System.out.println("Não temos como descer tudo isso.");
        }

    }

}

File: Python/Alura/pythonOO/Ex04/elevador.py
class Elevador:
    def __init__(self, total_andares, lotacao):
        self.__andar_atual = 0
        self.__lotacao_atual = 0
        self.__total_andares = total_andares
        self.__lotacao = lotacao

    def entrar(self, pessoas):
        if (self.__lotacao_atual + pessoas) > self.__lotacao:
            print("Quantidade de pessoas excede a lotação máxima do elevador.")
        elif (self.__lotacao_atual + pessoas) == self.__lotacao:
            self.__lotacao_atual += pessoas
            print(f"Entrou {pessoas} no elevador, o elevador está cheio.")
        else:
            self.__lotacao_atual += pessoas
            print(f"Entrou {pessoas} no elevador, o elevador está com {self.__lotacao_atual} pessoas.")

    def sair(self, pessoas):
        if (self.__lotacao_atual - pessoas) > 0:
            self.__lotacao_atual -= pessoas
            print(f"Saiu {pessoas} pessoas do elevador, o elevador está com {self.__lotacao_atual} pessoas.")
        elif (self.__lotacao_atual - pessoas) == 0:
            self.__lotacao_atual -= pessoas
            print(f"Saiu {pessoas} pessoas do elevador, o elevador está vazio.")
        else:
            print(f"Não há {pessoas} pessoas no elevador, há {self.__lotacao_atual}.")

    def subir(self, andares):
        if (andares + self.__andar_atual) > self.__total_andares:
            print(f"Elevador não pode subir {andares} andares.")
        elif (andares + self.__andar_atual) <= self.__total_andares:
            self.__andar_atual += andares
            print(f"O elevador subiu para o {self.__andar_atual}° andar.")

    def descer(self, andares):
        if self.__andar_atual - andares < 0:
            print(f"O elevador não pode descer {andares} andares.")
        elif self.__andar_atual - andares > 0:
            self.__andar_atual -= andares
            print(f"O elevador desdeu {andares} andares, e está no {self.__andar_atual}° andar.")
        elif self.__andar_atual - andares == 0:
            self.__andar_atual -= andares
            print(f"O elevador desdeu {andares} andares, e está no térreo.")
